host/xtest/pkcs11_1000.c

```diff
@@ -50,6 +50,75 @@ static void xtest_tee_test_1000(ADBG_Case_t *c)
 ADBG_CASE_DEFINE(pkcs11, 1000, xtest_tee_test_1000,
 		"Initialize and close Cryptoki library");
 
+static void print_mechanism_info(CK_MECHANISM_TYPE mecha_type,
+				 CK_MECHANISM_INFO_PTR mecha_info)
+{
+	/* 1024-byte should be enough, if not truncates  */
+	char msg[1024] = { };
+	size_t pos = 0;
+	size_t n = 0;
+
+	pos = snprintf(msg, sizeof(msg), "%-30s Key size [%03lu %03lu],",
+		       ckm2str(mecha_type), mecha_info->ulMinKeySize,
+		       mecha_info->ulMaxKeySize);
+
+	if (pos >= sizeof(msg)) {
+		Do_ADBG_Log("| Error: internal short buffer");
+		return;
+	}
+
+	if (!mecha_info->flags) {
+		Do_ADBG_Log("| %s all flags down", msg);
+		return;
+	}
+
+	pos += snprintf(msg + pos, sizeof(msg) - pos, " flags:");
+
+	for (n = 0; n < 32; n++) {
+		if (!(mecha_info->flags & (1UL << n)))
+			continue;
+
+		if (pos >= sizeof(msg))
+			break;
+
+		pos += snprintf(msg + pos, sizeof(msg) - pos,
+				" %s", mecha_ckf2str(1UL << n));
+	}
+
+	if (pos >= sizeof(msg))
+		Do_ADBG_Log("| Error: internal short buffer");
+	else
+		Do_ADBG_Log("| (%zu) %s", pos, msg);
+}
+
+static int check_mechanism_list(ADBG_Case_t *c, CK_SLOT_ID slot,
+				CK_MECHANISM_TYPE_PTR mecha_types,
+				CK_ULONG mecha_count)
+{
+	size_t n = 0;
+
+	/* Verbose output on high levels */
+	if (level)
```

Comment:
Level is not log level, it's how much testing is done. See xtest_tee_test_4006() for instance.

Reply:
I agree. But there is no verbosity level defined. I assume test level > 0 means a bit more than the min testing hence adding a trace is not that noisy.

Reply:
I'll discard this log stuff.